iib/workers/tasks/build.py

```diff
@@ -720,6 +725,69 @@ def _verify_index_image(
         )
 
 
+def _add_property_to_index(db_path, property):
+    """
+    Add a property to the index.
+
+    :param str db_path: path to the index database
+    :param dict property: a dict representing a property to be added to the index.db
+    """
+    insert = (
+        'INSERT INTO properties '
+        '(type, value, operatorbundle_name, operatorbundle_version, operatorbundle_path) '
+        'VALUES (?, ?, ?, ?, ?);'
+    )
+    con = sqlite3.connect(db_path)
+    # Insert property
+    con.execute(
+        insert,
+        (
+            property['type'],
+            property['value'],
+            property['operatorbundle_name'],
+            property['operatorbundle_version'],
+            property['operatorbundle_path'],
+        ),
+    )
+    con.commit()
+    con.close()
+
+
+def _requires_max_ocp_version(bundle):
+    """
+    Check if the bundle requires the olm.maxOpenShiftVersion property.
+
+    This property is required for bundles using deprecated APIs that don't already
+    have the olm.maxOpenShiftVersion property set.
+
+    :param str bundle: a string representing the bundle pull specification
+    :returns bool:
+    """
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1',
+        '--image-builder',
+        'none',
+    ]
+    result = run_cmd(cmd, strict=False)
```

Comment:
Could you explain why the command is allowed to fail?

Reply:
https://github.com/release-engineering/iib/pull/260#discussion_r664133524

Reply:
okay, but what if the operator-sdk command chokes on something and fails? Is that a tolerable state?